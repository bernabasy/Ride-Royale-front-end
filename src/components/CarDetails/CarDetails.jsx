import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import axios from 'axios';
import styles from './style.css';

const CarDetails = () => {
  const { id } = useParams();
  const [car, setCar] = useState({});

  useEffect(() => {
    axios
      .get(`http://127.0.0.1:3000/api/v1/cars/${id}`)
      .then((response) => setCar(response.data.car));
  }, []);

  return (
    <>
      <h2 className={styles.title}>CAR DETAILS</h2>
      <div className={styles.detailContainer}>
        <div className={styles.car}>
          <img src={car.picture} alt={car.model} />
        </div>
        <div className={styles.detail}>
          <h2>{car.model}</h2>
          <div className={styles.company}>
            <p>Company Name:</p>
            <p>{car.make}</p>
          </div>
          <div className={styles.price}>
            <p>Price Per Day: </p>
            <p className={styles.priceColor}>
              $
              {car.price}
            </p>
          </div>
          <button type="button" className={styles.reservebtn}>
            Reserve
          </button>
        </div>
      </div>
    </>
  );
};
export default CarDetails;
