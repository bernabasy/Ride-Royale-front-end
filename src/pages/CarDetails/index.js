import CarDetails from '../../components/CarDetails/CarDetails';
import Layout from '../../components/Layout/Layout';

const CarDetailPage = () => (
  <Layout>
    <CarDetails />
  </Layout>
);

export default CarDetailPage;
