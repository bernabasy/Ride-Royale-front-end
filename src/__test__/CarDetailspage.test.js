import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import CarDetails from '../Components/CarDetails/CarDetails';
import renderer from 'react-test-renderer';

it('Check the component', () => {
  const tree = renderer.create(
      <CarDetails />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
