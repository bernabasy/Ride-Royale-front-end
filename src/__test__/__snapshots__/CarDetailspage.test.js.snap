// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the component 1`] = `
Array [
  <h2
    className="title"
  >
    CAR DETAILS
  </h2>,
  <div
    className="detailContainer"
  >
    <div
      className="car"
    >
      <img />
    </div>
    <div
      className="detail"
    >
      <h2 />
      <div
        className="company"
      >
        <p>
          Company Name:
        </p>
        <p />
      </div>
      <div
        className="price"
      >
        <p>
          Price Per Day: 
        </p>
        <p
          className="priceColor"
        >
          $
        </p>
      </div>
      <button
        className="reservebtn"
        type="button"
      >
        Reserve
      </button>
    </div>
  </div>,
]
`;
